packages/flutter/lib/src/widgets/list_wheel_scroll_view.dart

```diff
@@ -0,0 +1,713 @@
+/// children must be the same size along the scrolling axis.
+///
+/// The children starts at the middle instead of at the top of the viewport.
+///
+/// Scrolling also ends with the last child at the middle instead of at the
+/// bottom of the viewport.
+///
+/// The children are also rendered as if on a wheel instead of on a plane.
+class ListWheelScrollView extends StatelessWidget {
+  /// Creates a box in which children are scrolled on a wheel.
+  const ListWheelScrollView({
+    Key key,
+    this.controller,
+    this.physics,
+    this.diameterRatio: 2.0,
+    this.perspective: 0.003,
+    @required this.itemExtent,
+    this.clipToSize: true,
+    this.renderChildrenOutsideViewport: false,
+    @required this.children,
+  }) : assert(diameterRatio != null && diameterRatio > 0.0),
+       assert(
+         perspective != null && perspective >= 0.0 && perspective < 0.01,
+         'A perspective too high will be clipped in the z axis and'
+         'un-renderable. Choose a value between 0 and 0.01.'
+       ),
+       assert(itemExtent != null && itemExtent > 0.0),
+       assert(clipToSize != null),
+       assert(renderChildrenOutsideViewport != null),
+       assert(
+         !renderChildrenOutsideViewport || !clipToSize,
+         'Cannot renderChildrenOutsideViewport and clipToSize since children'
+         'rendered outside will be clipped anyway, leading to gratuitous waste'
+       ),
+       super(key: key);
+
+  /// An object that can be used to control the position to which this scroll
+  /// view is scrolled.
+  ///
+  /// A [ScrollController] serves several purposes. It can be used to control
+  /// the initial scroll position (see [ScrollController.initialScrollOffset]).
+  /// It can be used to control whether the scroll view should automatically
+  /// save and restore its scroll position in the [PageStorage] (see
+  /// [ScrollController.keepScrollOffset]). It can be used to read the current
+  /// scroll position (see [ScrollController.offset]), or change it (see
+  /// [ScrollController.animateTo]).
+  final ScrollController controller;
+
+  /// How the scroll view should respond to user input.
+  ///
+  /// For example, determines how the scroll view continues to animate after the
+  /// user stops dragging the scroll view.
+  ///
+  /// Defaults to matching platform conventions.
+  final ScrollPhysics physics;
+
+  /// {@macro flutter.widgets.wheelList.diameterRatio}
+  ///
+  /// Defaults to an arbitrary but aesthetically sane number of 2.0.
+  ///
+  /// Must not be null and must be positive.
+  final double diameterRatio;
+
+  /// Perspective of the cylindrical projection.
+  ///
+  /// A number between 0.01 and 0 where 0 means looking at the cylinder from
+  /// infinitely far with an infinitely small field of view and 1 means looking
+  /// at the cylinder from infinitely close with an infinitely large field of
+  /// view (and also un-renderable).
```

Comment:
(and also un-renderable) -> (which cannot be rendered)

Reply:
Done